core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -163,8 +163,9 @@ protected BikeCommonFlagEncoder( int speedBits, double speedFactor, int maxTurnC
         setHighwaySpeed("living_street", 6);
         setHighwaySpeed("steps", PUSHING_SECTION_SPEED / 2);
 
-        setHighwaySpeed("cycleway", 18);
-        setHighwaySpeed("path", 12);
+        final int CYCLEWAY_SPEED = 18;  // Make sure cycleway and path use same speed value, see #634
+        setHighwaySpeed("cycleway", CYCLEWAY_SPEED);
+        setHighwaySpeed("path", CYCLEWAY_SPEED);
```

Comment:
Sorry. I overlooked these question here. This is a good one. But the answer is not that easy.
The explicit highway speed for path is actually not used at all, because of these lines on line 441 in BikeCommonFlagEncoder.java
```
                   if (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official"))
                      speed = getHighwaySpeed("cycleway");
```
But because ot the check in line 428 
`if (speed > PUSHING_SECTION_SPEED)`
it is important that it is assigned with a highway speed > PUSHING_SECTION_SPEED. 
Maybe there might be a simpler and more elegant solution, but I didn't one. Please try to play around with it, then you'll see what I mean.